src/request_builder.rs

```diff
@@ -246,11 +244,10 @@ impl Future for RequestBuilder {
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         if self.fut.is_none() {
             let req = self.req.take().unwrap();
-
             if let Some(client) = &self.client {
                 self.fut = Some(client.send(req))
             } else {
-                let client = Client::new_shared();
+                let client = Client::new_shared_or_panic();
```

Comment:
It might be too much cfg statements but `RequestBuilder` would _ideally_ not impl future if there isn't a default.

Reply:
That would interfere with this usage:

```rust
/// surf with no default features
Client::with_http_client(Arc::new(some_http_client)) // Client
    .get("http://example.com") // RequestBuilder
    .await // using Future on RequestBuilder
```